### PR TITLE
Compose packages

### PR DESCRIPTION
**Pull request by kevinemoore**
Allow package builds to include other packages and sub-packages as nodes.

### CODE REVIEW

**Comment by asah** [comment]
@kevinemoore  JFYI I'm getting an error...

build.yml
-------------------------------------------
contents:
  README:
    file: README.md
  build:
    file: build.yml
  snli:
    package: asah/snli

$ ls -lR
total 24
-rw-r--r--  1 asah  staff  5828 Aug 29  2015 README.md
-rw-r--r--  1 asah  staff   100 Feb 27 13:53 build.yml

$ cat build.yml
contents:
  README:
    file: README.md
  build:
    file: build.yml
  snli:
    package: asah/snli

$ quilt build examples/snli ./build.yml
Inferring 'transform: id' for README.md
Registering ./README.md...
Inferring 'transform: id' for build.yml
Registering ./build.yml...
Built examples/snli successfully.

$ quilt push --public examples/snli
Fetching upload URLs from the registry...
{'type': 'ROOT', 'children': {}} is not valid under any of the given schemas

**Comment by asah** [comment]
My issue is resolved.  thanks!

File: compiler/quilt/test/test_build.py [hunk 1]
```diff
@@ -357,3 +357,83 @@ def test_package_contains(self):
...
+        assert simple.foo().equals(compose1.from_simple_foo())
+
+    def test_compose_package_not_found(self):
+        mydir = pathlib.Path(os.path.dirname(__file__))
+        buildfile = mydir / 'build_simple.yml'
+        command.build('test/simple', str(buildfile))
+
+        missing_dep_build = {
+            'contents': {
+            'foo': {
```
Comment (dimaryaz) [inline compiler/quilt/test/test_build.py]: Indentation

File: compiler/quilt/tools/build.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def _path_hash(path, transform, kwargs):
     return digest_string(srcinfo)
 
 def _is_internal_node(node):
-    is_leaf = not node or node.get(RESERVED['file'])
+    is_leaf = not node or node.get(RESERVED['file']) or node.get(RESERVED['package'])
```
Comment (dimaryaz) [inline compiler/quilt/tools/build.py]: Make sure this still works after the #414 merge.

File: compiler/quilt/tools/command.py [hunk 3]
```diff
@@ -91,53 +88,35 @@ def __init__(self, message, response):
...
+            team, owner, pkg = parse_package_util(name, allow_subpath)
+            subpath = None
     except ValueError:
         pkg_format = '[team:]owner/package_name/path' if allow_subpath else '[team:]owner/package_name'
         raise CommandException("Specify package as %s." % pkg_format)
 
-    try:
-        PackageStore.check_name(team, owner, pkg, subpath)
-    except StoreException as ex:
+    try:
```
Comment (dimaryaz) [inline compiler/quilt/tools/command.py]: Whitespace

File: compiler/quilt/tools/build.py [hunk 4]
```diff
@@ -164,88 +164,109 @@ def _build_node(build_dir, package, name, node, fmt, target='pandas', checks_con
...
-
-            # Check to see that cached objects actually exist in the store
-            # TODO: check for changes in checks else use cache
-            # below is a heavy-handed fix but it's OK for check builds to be slow  
-            if not checks and cachedobjs and all(os.path.exists(store.object_path(obj)) for obj in cachedobjs):
-                # Use existing objects instead of rebuilding
-                package.save_cached_df(cachedobjs, name, rel_path, transform, target, fmt)
+       
+        include_package = node.get(RESERVED['package'])
+        if include_package:
```
Comment (dimaryaz) [inline compiler/quilt/tools/build.py]: Maybe report an error if both `package` and `file` are present?

**Pull request merged**